WordPress/gutenberg
# Migrate navigation editor selected menu to preferences store

Part of https://github.com/WordPress/gutenberg/issues/31965

## What?
Migrates the navigation editors `selectedMenuId` persisted state over to the preferences store.

## Why?
As outlined in https://github.com/WordPress/gutenberg/issues/31965, the goal is to move persisted data out of our various stores to a centralized preferences store.

The navigation editor is still experimental, a bit broken and is not being actively developed, but this is a very easy migration so no reason not to do it.

## How?
I've removed the selector, action and reducer that previously managed this state. The navigation editor already uses a `useSelectedMenuId` hook for this feature, so it was simple enough to select from and dispatch to the preferences store within this hook.

In other PRs I've also added code to migrate the local storage data for each preference, but given the navigation editor's current state, there won't be anyone actively using it, so I don't think that's actively necessary this time.

## Testing Instructions
1. Enable the experimental Navigation Editor on the Gutenberg > Experiments page
2. If you only have one or no menus, create some using the New Menu button
3. Using the little dropdown in the middle of the top bar, switch menu, the editor should change between menus
4. Reload the page, the editor should open up the last menu.
